Assets/SEE/UI/PropertyDialog/UserPropertyDialog.cs

```diff
@@ -0,0 +1,162 @@
+﻿using SEE.Controls;
+using SEE.UI.Notification;
+using System;
+using SEE.Utils;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace SEE.UI.PropertyDialog
+{
+    internal class UserPropertyDialog
+    {
+        /// <summary>
+        /// Callback to be called when this dialog closes.
+        /// </summary>
+        public delegate void OnClosed();
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="networkConfig">the network configuration to be manipulated by this dialog</param>
+        /// <param name="callBack">delegate to be called when this dialog is closed</param>
+        public UserPropertyDialog(Net.Network networkConfig, OnClosed callBack = null)
+        {
+            this.networkConfig = networkConfig;
+            this.callBack = callBack;
+        }
+
+        /// <summary>
+        /// Delegate to be called when this dialog is closed. May be null in which case
+        /// nothing is called.
+        /// </summary>
+        private OnClosed callBack;
+
+        /// <summary>
+        /// The dialog used to manipulate the node.
+        /// </summary>
+        private GameObject dialog;
+
+        /// <summary>
+        /// The dialog property for the playername.
+        /// </summary>
+        private StringProperty playerName;
+
+        /// <summary>
+        /// The dialog for the user input.
+        /// </summary>
+        private PropertyDialog propertyDialog;
+
+        /// <summary>
+        /// Event triggered when the user presses the OK button. Clients can
+        /// register on this event to receive a notification when this happens.
+        /// </summary>
+        public readonly UnityEvent OnConfirm = new();
+
+        /// <summary>
+        /// Event triggered when the user presses the Cancel button. Clients can
+        /// register on this event to receive a notification when this happens.
+        /// </summary>
+        public readonly UnityEvent OnCancel = new();
```

Comment:
I think these can be set to `private`.

Reply:
I'm not sure if these are maybe used from outside anywhere in `NetworkPropertyDialog`. Please check that first, and if it isn't the case, then you can set those ones to private too. I think the ones in `UserPropertyDialog` aren't used from outside at least, so these can be set to private in any case.